stacker/actions/diff.py

```diff
@@ -0,0 +1,146 @@
+import logging
+
+from .. import exceptions
+from ..plan import COMPLETE, SKIPPED, Plan
+from . import build
+import difflib
+import json
+import sys
+
+logger = logging.getLogger(__name__)
+
+
+class Action(build.Action):
+    """ Responsible for diff'ing CF stacks in AWS and on disk
+
+    Generates the build plan based on stack dependencies (these dependencies
+    are determined automatically based on references to output values from
+    other stacks).
+
+    The plan is then used to pull the current CloudFormation template from
+    AWS and compare it to the generated templated based on the current
+    config.
+    """
+
+    def _diff_parameters(self, old_params, new_params):
+        """Compares the old vs. new parameters and returns a 'diff'
+
+        Args:
+            old_params(dict): old paramters
+            new_params(dict): new parameters
+
+        Returns:
+            int: number of changed entries
+            string: "diff" of all parameters and how they changed
+        """
+        output = []
+        count = 0
+        for k, v in old_params.iteritems():
```

Comment:
i'm personally not a fan of these variable names. i would go with the more verbose option of:
```
for old_key, new_value in old_params.iteritems():
```

then the logic below, ie:

```
new_value = new_params.get(old_key, "_Undef_")
```

is much clearer.